File: 1) INTRODUCCION_A_POO/Ejercicio-3/PYTHON/Producto.py
class Producto:
    def __init__(self, nombre, precio, stock):
        self.nombre = nombre
        self.precio = precio
        self.stock = stock

    def vender(self, cantidad):
        if self.stock >= cantidad:
            self.stock -= cantidad

    def reabastecer(self, cantidad):
        self.stock += cantidad

    def __str__(self):
        return f"Producto: {self.nombre} | Precio: ${self.precio} | Stock: {self.stock}"

if __name__ == "__main__":
    frijoles = Producto("Frijoles", 1.99, 50)

    print("---  PRODUCTO  ---")
    print("Estado Inicial:")
    print(frijoles)

    frijoles.vender(15)
    print("\nDespués de vender 15 unidades:")
    print(frijoles)

    frijoles.reabastecer(75)
    print("\nDespués de reabastecer 75 unidades:")
    print(frijoles)

    frijoles.vender(300)
    print("\nIntentando vender 300 (sin stock suficiente):")
    print(frijoles)

File: 1) INTRODUCCION_A_POO/Ejercicio-3/JAVA/Producto.java
public class Producto {
    String nombre;
    double precio;
    int stock;

    public Producto(String nombre, double precio, int stock) {
        this.nombre = nombre;
        this.precio = precio;
        this.stock = stock;
    }

    public void vender(int cantidad) {
        if (this.stock >= cantidad) {
            this.stock -= cantidad;
        }
    }

    public void reabastecer(int cantidad) {
        this.stock += cantidad;
    }

    @Override
    public String toString() {
        return "Producto: " + nombre + " | Precio: $" + precio + " | Stock: " + stock;
    }

    public static void main(String[] args) {
        Producto arroz = new Producto("Arroz", 3.25, 100);

        System.out.println("--- PRUEBA DE CLASE PRODUCTO (JAVA) ---");
        System.out.println("Estado Inicial:");
        System.out.println(arroz);

        arroz.vender(20);
        System.out.println("\nDespués de vender 20 unidades:");
        System.out.println(arroz);

        arroz.reabastecer(50);
        System.out.println("\nDespués de reabastecer 50 unidades:");
        System.out.println(arroz);

        arroz.vender(200);
        System.out.println("\nIntentando vender 200 (sin stock suficiente):");
        System.out.println(arroz);
    }
}
